### PR TITLE
use TIMER_CHANNEL_NUMOF instead of TIMER_CHANNELS

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

In #13900 we agreed to make `TIMER_CHANNEL_NUMOF` the new common name for the number of timer channels a platform provides.

So let's start by renaming all instances of `TIMER_CHANNELS`.

### Testing procedure

Murdock should confirm I didn't miss anything.

### Issues/PRs references

alternative to #13900

### CODE REVIEW

File: cpu/esp32/include/periph_cpu.h [hunk 1]
```diff
@@ -507,11 +507,11 @@ typedef struct {
 #ifdef MODULE_ESP_HW_COUNTER
 /** hardware ccount/ccompare registers are used for timer implementation */
 #define TIMER_NUMOF     (2)
-#define TIMER_CHANNELS  (1)
+#define TIMER_CHANNEL_NUMOF  (1)
```
Comment (basilfx) [inline cpu/esp32/include/periph_cpu.h:510]: Nit-picky: alignment.

File: cpu/esp32/include/periph_cpu.h [hunk 2]
```diff
@@ -507,11 +507,11 @@ typedef struct {
...
 #else
 /** hardware timer modules are used for timer implementation (default) */
 #define TIMER_NUMOF     (3)
-#define TIMER_CHANNELS  (1)
+#define TIMER_CHANNEL_NUMOF  (1)
```
Comment (basilfx) [inline cpu/esp32/include/periph_cpu.h:514]: Nit-picky: alignment.

Comment (benpicco) [reply cpu/esp32/include/periph_cpu.h]: aligned.

**Comment by basilfx** [comment]
ACK, please squash.

**Pull request merged**